### PR TITLE
Adjust default SMT settings for RPC based proofs

### PR DESCRIPTION
**Pull request by ehildenb**
Blocked on: https://github.com/runtimeverification/evm-semantics/pull/1646

This attempts to up the SMT timeout to find the smallest viable one for getting things merged faster.

This PR:

- Adds a new argument parser for `smt_args`, which contains:
  - `--smt-timeout`: control how long smt queries take.
  - `--smt-retry-limit`: control how many times to retry the SMT solver.
- Threads through these CLI arguments to all places that invoke the backend via `KCFGExplore`.
- Removes `find_free_port()` and relies on upstream port finding, which does not have the race condition present here.

### CODE REVIEW

**Comment by ehildenb** [comment]
We should still get https://github.com/runtimeverification/evm-semantics/pull/1635 sorted, because @jberthold is reporting that every invocation of the new booster backend we are accumulating those Ceil constraints, nested on each other, when we really only need them the first call to generate the new conditions. The booster backend doesn't have a ceil simplifier, or a simplifier at all, so it results in the term it sees growing indefinitely every time the execute endpoint is called.

File: kevm-pyk/src/kevm_pyk/kevm.py [hunk 1]
```diff
@@ -60,6 +60,10 @@ def __init__(
             bug_report=bug_report,
         )
 
+    @staticmethod
+    def kore_rpc_command() -> Iterable[str]:
+        return ('kore-rpc', '--smt-timeout', '500', '--smt-retry-limit', '4')
```
Comment (anvacaru) [inline kevm-pyk/src/kevm_pyk/kevm.py:65]: Is the `500` here a typo?

File: kevm-pyk/src/kevm_pyk/foundry.py [hunk 2]
```diff
@@ -343,7 +345,9 @@ def foundry_prove(
             target_term = KDefinition__expand_macros(foundry.kevm.definition, target_term)
             kcfg.replace_node(kcfg.get_unique_target().id, CTerm(target_term))
             if simplify_init:
-                with KCFGExplore(foundry.kevm, port=find_free_port(), bug_report=br) as kcfg_explore:
+                with KCFGExplore(
+                    foundry.kevm, port=find_free_port(), bug_report=br, kore_rpc_command=kore_rpc_command
```
Comment (tothtamas28) [inline kevm-pyk/src/kevm_pyk/foundry.py:349]: Added SMT options to `__init__`: https://github.com/runtimeverification/pyk/pull/276

Comment (tothtamas28) [reply kevm-pyk/src/kevm_pyk/foundry.py:349]: Also, `find_free_port` can be dropped now: https://github.com/runtimeverification/pyk/pull/262

File: kevm-pyk/src/kevm_pyk/kevm.py (hunk 1 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/kevm.py]: Yes, fixed.

File: kevm-pyk/src/kevm_pyk/foundry.py (hunk 2 above)
Comment (ehildenb) [reply kevm-pyk/src/kevm_pyk/foundry.py]: Done.

**Comment by ehildenb** [comment]
Here: https://github.com/runtimeverification/evm-semantics/pull/1640 We actually end up updating K to the point where we get proper "SMT failure" messages, so now we'll try and up the timeout to see if those messages go away.

Another effort is here: https://github.com/runtimeverification/evm-semantics/pull/1635 to make it so we don't need the solver as much anyway.

We'll re-run CI here a few times to make sure it's actually consistently passing, then reduce the increased timeout. "Successful" means "did not produce any SMT errors".

### 500ms

- Successful 3d1c27d GH13 46m: https://github.com/runtimeverification/evm-semantics/actions/runs/4453881500/jobs/7822775725
- Successful 31e7ea0 GH10 64m: https://github.com/runtimeverification/evm-semantics/actions/runs/4455757069/jobs/7825866023
- Successful 31e7ea0 GH15 75m: https://github.com/runtimeverification/evm-semantics/actions/runs/4455757069/jobs/7826356983 (race condition on opening socket)
- Successful 2e4e6e4 GH15 74m: https://github.com/runtimeverification/evm-semantics/actions/runs/4456726625/jobs/7827443074
- Successful 2e4e6e4 GH13 46m: https://github.com/runtimeverification/evm-semantics/actions/runs/4456726625/jobs/7827996896

### 250ms

- Successful 9b96224 GH13 46m: https://github.com/runtimeverification/evm-semantics/actions/runs/4457362759/jobs/7828463559
- Successful 9b9224 GH10 64m: https://github.com/runtimeverification/evm-semantics/actions/runs/4457362759/jobs/7828923280 (some docker error, tests passed)
- Successful 9b96224 GH13 46m: https://github.com/runtimeverification/evm-semantics/actions/runs/4457362759/jobs/7829658784
- Successful 9b96224 GH10 64m: https://github.com/runtimeverification/evm-semantics/actions/runs/4457362759/jobs/7830151570
- Successful 9b96224 GH10 64m: https://github.com/runtimeverification/evm-semantics/actions/runs/4457362759/jobs/7830652493

### 125ms

- Successful 5cb09d6 GH15 75m: https://github.com/runtimeverification/evm-semantics/actions/runs/4459074665/jobs/7831383287
- Successful 5cb09d6 GH10 65m: https://github.com/runtimeverification/evm-semantics/actions/runs/4459074665/jobs/7831853323
- Successful bdd1ff1 GH16 64m: https://github.com/runtimeverification/evm-semantics/actions/runs/4459703797/jobs/7832406077
- Successful bdd1ff1 GH10 65m: https://github.com/runtimeverification/evm-semantics/actions/runs/4459703797/jobs/7832785869
- Mostly successful 8be4355 GH10 66m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461460010/jobs/7835381088 (Haskell backend proofs failed due to resource exhaustion)

### Default: --smt-timeout 125 --smt-retry-limit 4

- Successful 6031a0e GH16 50m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461777091/jobs/7836009228 (failure on a Nix stage)
- Successful 6031a0e GH16 48m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461777091/jobs/7836962660
- Successful 6031a0e GH10 67m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461777091/jobs/7837466151
- Successful 6031a0e GH10 66m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461777091/jobs/7837920365
- Successful 6031a0e GH10 67m: https://github.com/runtimeverification/evm-semantics/actions/runs/4461777091/jobs/7839495404

**Review by jberthold — APPROVED** [review]
LGTM  :thread: :sewing_needle:

**Pull request merged**